services/121-service/src/transaction-job-processors/transaction-job-processors.service.ts

```diff
@@ -134,24 +145,20 @@ export class TransactionJobProcessorsService {
           )?.value as string, // This must be a string. If it is not, the intersolve API will return an error (maybe).
         });
     } catch (error) {
-      await this.createTransaction({
-        amount: input.transactionAmount,
-        registration: registration,
+      await this.processTransactionResult({
+        jobInput: input,
+        calculatedTranserAmount: input.transactionAmount, // TODO: STORE THE CALCULATED AMOUNT HERE AS THIS IS USED ON RETRY AND SHOWN IN PORTAL
         financialServiceProviderId: financialServiceProvider.id,
-        programId: input.programId,
-        paymentNumber: input.paymentNumber,
-        userId: input.userId,
+        registration,
+        oldRegistration,
+        isRetry: input.isRetry,
         status: StatusEnum.error,
-        errorMessage: `An error occured: ${error}`,
+        errorText: error?.message, // TODO: THIS IS A GENEARAL stack trace catch and i think the error handling should be more specific and tested
       });
-      await this.updatePaymentCountAndStatusInRegistration(
-        registration,
-        input.programId,
-      );
-      throw new Error(error);
+      return;
```

Comment:
Return instead of throw error, to not pollute our logs

Reply:
Indeed. Creating a failed transaction is "normal behavior" and should not throw an error.